pandas/core/strings.py

```diff
@@ -472,9 +472,17 @@ def str_extract(arr, pat, flags=0):
     from pandas.core.index import Index
 
     regex = re.compile(pat, flags=flags)
-    # just to be safe, check this
+
     if regex.groups == 0:
         raise ValueError("This pattern contains no groups to capture.")
+    
+    if isinstance(arr, Index):
+        if 1 < regex.groups:
+            raise ValueError("only one regex group is supported with Index")
```

Comment:
Seems to me that there is no reason we should impose a limit of only one regex group for indices. This feature comes from https://github.com/pydata/pandas/issues/9980 which explains that they want extract to return an Index when it is called on an Index. In that case it seems like extract with one group should return an Index, and with several groups it should return a MultiIndex (not an error as in current implementation). what do you think?